pulsar-broker-common/src/main/java/org/apache/pulsar/broker/web/AuthenticationFilter.java

```diff
@@ -76,8 +77,15 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
                 // not sasl type, return role directly.
                 String role = authenticationService.authenticateHttpRequest((HttpServletRequest) request);
```

Comment:
Since this PR is added the appropriate headers by default, I think we should clean up and possibly deprecate this `authenticateHttpRequest` method. If you look at its internal definition, it branches based on `"X-Pulsar-Auth-Method-Name"`. Note also that the current design of this PR will result in double verification of the authentication data, which should be avoided.